### PR TITLE
Ensure that ImageDataGenerator.flow() generates stratified training a…

### PR DESCRIPTION
**Pull request by mfridental**
…nd test sets even if the passed numpy arrays are sorted by class

### Summary
A small improvement that fixes a problem that could take days to debug. 

In case the initial numpy arrays used for ImageDataGenerator.flow() are sorted by class, which can be the case if the user's code is reading images class by class, there is a high chance that the class frequencies in the resulting training and tests will differ. In the worst case, the training set will contain only samples of one class and the test set samples of another class, which would lead to poor model generalization.

The proposed patch fixes the problem by shuffling numpy arrays before they are split into training and validation.

### Related Issues
The issue happened to me at work. I've fixed it rather than opening an issue.

### PR Overview

- [n] This PR requires new unit tests [y/n] (make sure tests are included) (OLD TEST EXTENDED TO COVER THE FIX)
- [n] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [y] This PR is backwards compatible [y/n]
- [n] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

**Comment by mfridental** [comment]
Will fix it and make another pull request
